src/fermilib/utils/_jellium_trotter_error.py

```diff
@@ -0,0 +1,461 @@
+    for i in range(n_qubits):
+        for j in range(n_qubits):
+            two_number_action = ((i, 1), (j, 1), (i, 0), (j, 0))
+            if hamiltonian.terms.get(two_number_action):
+                number_number_appearances[i] += 1
+                number_number_appearances[j] += 1
+
+    for i in range(n_qubits):
+        for j in range(1, n_qubits):
+            index_left = min(i, (i + j) % n_qubits)
+            index_right = max(i, (i + j) % n_qubits)
+
+            hopping_action1 = ((index_left, 1), (index_right, 0))
+            hopping_action2 = ((index_right, 1), (index_left, 0))
+            two_number_action = ((index_right, 1), (index_left, 1),
+                                 (index_right, 0), (index_left, 0))
+            left_number_action = ((index_left, 1), (index_left, 0))
+            right_number_action = ((index_right, 1), (index_right, 0))
+
+            hopping_operator = (
+                FermionOperator(hopping_action1,
+                                hamiltonian.terms.get(hopping_action1, 0.0)) +
+                FermionOperator(hopping_action2,
+                                hamiltonian.terms.get(hopping_action2, 0.0)))
+            hopping_operator.compress()
+            hopping_operator /= 2.0
+
+            number_operator = FermionOperator(two_number_action,
+                                              hamiltonian.terms.get(
+                                                  two_number_action, 0.0))
+            if not number_operator.isclose(zero):
+                number_operator += (
+                    FermionOperator(left_number_action,
+                                    hamiltonian.terms.get(
+                                        left_number_action, 0.0) /
+                                    number_number_appearances[index_left]) +
+                    FermionOperator(right_number_action,
+                                    hamiltonian.terms.get(
+                                        right_number_action, 0.0) /
+                                    number_number_appearances[index_right]))
+            number_operator.compress()
+            number_operator /= 2.0
+
+            if not hopping_operator.isclose(zero):
+                ordered_terms.append(hopping_operator)
+                ordered_indices.append(set((index_left, index_right)))
+                ordered_is_hopping_operator.append(True)
+            if not number_operator.isclose(zero):
+                ordered_terms.append(number_operator)
+                ordered_indices.append(set((index_left, index_right)))
+                ordered_is_hopping_operator.append(False)
+
+    return (ordered_terms, ordered_indices, ordered_is_hopping_operator)
+
+
+def ordered_jellium_terms_no_info(
+        grid_length, dimension=3, wigner_seitz_radius=10.,
+        inverse_filling_fraction=2, spinless=True):
+    """Give terms from the jellium Hamiltonian in simulation algorithm order.
+
+    Gives a fairly arbitrary term ordering.
```

Comment:
The term ordering depends on converting a dictionary to a list, so it has no guaranteed order at this point.